python/taichi/lang/__init__.py

```diff
@@ -265,6 +270,20 @@ def supported_archs():
     return archs
 
 
+def arch_fallback(arch):
```

Comment:
This function should be named `adaptive_arch_selection` since it's not always falling back.